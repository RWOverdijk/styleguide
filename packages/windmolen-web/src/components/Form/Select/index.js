// @flow
import React, { type Node, Fragment } from 'react';
import styled from 'styled-components';
import { colors } from '../../../globals';
import Base from '../../Base';
import Icon  from '../../Icon';

type Props = {
  children: HTMLOptionsCollection
};

const StyledLabel = styled(Base.withComponent('label'))`
  color: ${colors.charcoalGray};
  margin-bottom: 5px;
  font-weight: bold;
  font-size: 18px;
`;

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
`;

const StyledSelectWrapper = styled.div`
  display: inline-flex;
  align-items: center;
  position: relative;
  overflow: hidden;
  height: 50px;
  box-shadow: 0 0 5px 0 rgba(51, 61, 71, 0.2);
  background-color: ${colors.white};
`;

const StyledIcon = styled(Icon)`
  pointer-events: none;
  position: absolute;
  right: 5px;
  transform: rotate(90deg);
`;

const StyledSelect = styled.select`
  background: transparent;
  border: none;
  font-size: 16px;
  height: 50px;
  padding: 9px 50px 9px 20px;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
`;

const Select = ({ children, label, id, ...props }: Props) => (
  <Wrapper>
    {label && (
      <StyledLabel for={id}>{label}</StyledLabel>
    )}
    <StyledSelectWrapper>
      <StyledSelect id={id} {...props}>
        {children}
      </StyledSelect>
      <StyledIcon name="arrow-right" fontSize="30px" />
    </StyledSelectWrapper>
  </Wrapper>
);

export default Select;
