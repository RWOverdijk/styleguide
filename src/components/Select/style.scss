/** @define Select */

@import "../../css/variables";
@import "../../css/mixins";

.Van-Select {
  position: relative;
  display: block;
  background-color: $color-background-bright;
  height: 40px;

  /* postcss-bem-linter: ignore */
  select {
    position: absolute;
    width: 100%;
    height: 100%;
    border: 0;
    -webkit-appearance: none;
    -moz-appearance: none;
    z-index: 3;
    background: transparent;
    padding: 0 15px;

    // https://stackoverflow.com/a/11603104
    color: #000;
    color: rgba(0, 0, 0, 0);
    text-shadow: 0 0 0 #000;

    /* postcss-bem-linter: ignore */
    &::-ms-expand {
      display: none; /* hide the default arrow in ie10 and ie11 */
    }
  }

  /* postcss-bem-linter: ignore */
  .Van-Icon {
    @include color-fill($color-grey);

    position: absolute;
    right: 20px;
    height: auto;
    width: 14px;
    top: 13px;
  }
}
