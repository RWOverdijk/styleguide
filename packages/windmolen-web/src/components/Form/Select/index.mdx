---
name: Select
menu: Forms
route: /Forms/Select
---

import { Playground, PropsTable } from 'docz';
import Wrapper from '../../Wrapper';
import Select from './';

# Select

## Props
<PropsTable of={Select} />

## Basic usage
<Playground wrapper={Wrapper}>
  <Select>
    <option value="female">Female</option>
    <option value="male">Male</option>
  </Select>
</Playground>
