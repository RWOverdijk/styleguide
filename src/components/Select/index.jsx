import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';
import arrowDownIcon from '../../assets/icons/fa/chevron-down.svg';

import './style.scss';

class Select extends PureComponent {
  render() {
    const { className, children, ...props } = this.props;

    return (
      <div className={classnames('Van-Select', className)} {...props}>
        <select>{children}</select>
        <Icon source={arrowDownIcon} />
      </div>
    );
  }
}

Select.defaultProps = {
  className: '',
  children: ''
};

Select.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

export default Select;
