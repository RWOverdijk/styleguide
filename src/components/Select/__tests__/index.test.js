import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../';

describe('<Select>', () => {
  it("renders Select", () => {
    const tree = renderer.create(
      <Select>
        <option>Option 1</option>
        <option>Option 2</option>
        <option>Option 3</option>
      </Select>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Select className="my-class">
        <option>Option 1</option>
        <option>Option 2</option>
        <option>Option 3</option>
      </Select>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Select data-id="0">
        <option>Option 1</option>
        <option>Option 2</option>
        <option>Option 3</option>
      </Select>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
