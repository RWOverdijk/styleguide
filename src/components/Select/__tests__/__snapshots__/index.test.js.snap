// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> renders Select 1`] = `
<div
  className="Van-Select"
>
  <select>
    <option>
      Option 1
    </option>
    <option>
      Option 2
    </option>
    <option>
      Option 3
    </option>
  </select>
</div>
`;

exports[`<Select> renders with specific attributes 1`] = `
<div
  className="Van-Select"
  data-id="0"
>
  <select>
    <option>
      Option 1
    </option>
    <option>
      Option 2
    </option>
    <option>
      Option 3
    </option>
  </select>
</div>
`;

exports[`<Select> renders with specific className 1`] = `
<div
  className="Van-Select my-class"
>
  <select>
    <option>
      Option 1
    </option>
    <option>
      Option 2
    </option>
    <option>
      Option 3
    </option>
  </select>
</div>
`;
